simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileBlockStreamManager.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.generator;
+
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
+import com.hedera.block.simulator.config.types.GenerationMode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+
+/** The block as file block stream manager. */
+public class BlockAsFileBlockStreamManager implements BlockStreamManager {
+
+    private static final System.Logger LOGGER =
+            System.getLogger(BlockAsFileBlockStreamManager.class.getName());
+
+    final String rootFolder;
+
+    final List<Block> blockList = new ArrayList<>();
```

Comment:
You can shorten the name from `blockList` to `blocks` since the type already communicates the data structure.

Reply:
✅  Done!